data/basal/cleanup.go

```diff
@@ -0,0 +1,65 @@
+package basal
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/mdblp/tide-whisperer-v2/v2/schema"
+	schemaV1 "github.com/tidepool-org/tide-whisperer/schema"
+)
+
+func CleanUpBasals(basals []schema.BasalBucket, loopModes []schemaV1.LoopModeEvent) []schema.BasalBucket {
+	var endBasalTime time.Time
+	sort.SliceStable(basals, func(i, j int) bool {
+		return basals[i].Day.Before(basals[j].Day)
+	})
+	cleanBuckets := []schema.BasalBucket{}
+	countBasal := 0
+	for _, basalDay := range basals {
+		sort.SliceStable(basalDay.Samples, func(i, j int) bool {
+			return basalDay.Samples[i].Timestamp.Before(basalDay.Samples[j].Timestamp)
+		})
+		cleanSamples := []schema.BasalSample{}
+		for _, basal := range basalDay.Samples {
+			guid := basal.Guid
+			// Filling time holes with 0 rate basals...
+			if !endBasalTime.IsZero() && basal.Timestamp.Sub(endBasalTime) >= time.Second {
+				sampleTime := endBasalTime
+				endBasalTime = basal.Timestamp
+				for _, loopMode := range schemaV1.GetLoopModeEventsBetween(sampleTime, &endBasalTime, loopModes) {
+					fmt.Printf("%v %v %v\n", sampleTime, endBasalTime, loopMode)
+					currentSample := getSample(guid, countBasal, 0, basal.Timezone, endBasalTime, loopMode)
+					cleanSamples = append(cleanSamples, currentSample)
+					countBasal++
+				}
+			}
+			// Checking basal overlap
+			// Reprocessing previous basal duration
+			if (endBasalTime.After(basal.Timestamp)) {
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
**💣 chore:** unnecessary parenthesis‏